NuGet/Get-BcNuGetPackage.ps1

```diff
@@ -1,101 +1,133 @@
-        $searchUrl = $searchResource.'@id' | Select-Object -First 1
-    }
-    catch {
-        throw (GetExtendedErrorMessage $_)
-    }
-    if (-not $searchUrl) {
-        Write-Host "Supported capabilities:"
-        $capabilities.resources.'@type' | ForEach-Object { Write-Host "- $_" }
-        throw "$nuGetServerUrl doesn't support SearchQueryService."
-    }
-    if (!$silent) {
-        Write-Host "Using $searchUrl"
-    }
-    try {
-        $searchResult = Invoke-RestMethod -UseBasicParsing -Method GET -Headers $headers -Uri "$searchUrl/?q=$packageName"
-        $count = $searchResult.data.count
-        Write-Host "$count matching packages found"
-        if ($count -gt 1) {
-            $searchResult.data | ForEach-Object { Write-Host "- $($_.id)" }
-        }
-        $packageMetadata = $searchResult.data | Where-Object { $_.id -eq $packageName }
-    }
-    catch {
-        throw (GetExtendedErrorMessage $_)
-    }
-
-    $tmpFolder = ''
-    if (-not $packageMetadata) {
+    function dump([string]$message) {
         if (!$silent) {
-            Write-Host "The package named $packageName wasn't found on $nuGetServerUrl"
+            Write-Host $message
         }
     }
-    else {
-        if (!$silent) {
-            Write-Host "Found Package $($packageMetadata.id) on $nuGetServerUrl"
-        }
 
-        $packageVersion = $packageMetadata.versions | Where-Object { [System.Version]$_.version -ge $version } | Sort-Object { [System.Version]$_.version } | Select-Object -Last 1
-        if ($packageVersion.'@id' -notlike 'https://*' -and (($searchUrl -like 'https://pkgs.dev.azure.com/*/v3/query2/') -or ($searchUrl -like 'https://*.pkgs.visualstudio.com/_packaging/*/nuget/v3/query2/'))) {
-            # Azure DevOps doesn't store URLs to package metadata in @id
-            $contentUrl = "$($searchUrl.Substring(0,$searchUrl.Length-10).Replace('/_packaging/','/_apis/packaging/feeds/'))packages/$($packageVersion.'@id')/versions/$($packageVersion.Version)/content"
-        }
-        else {
+    $bestmatch = $null
+    # Search all trusted feeds for the package
+    foreach($feed in (@([PSCustomObject]@{ "Url" = $nuGetServerUrl; "Token" = $nuGetToken; "Patterns" = @('*') })+$bcContainerHelperConfig.TrustedNuGetFeeds)) {
+        if ($feed -and $feed.Url) {
+            Dump "::group::Search NuGetFeed $($feed.Url)"
             try {
-                $package = Invoke-RestMethod -UseBasicParsing -Method GET -Headers $headers -Uri $packageVersion.'@id'
-                $contentUrl = $package.packageContent
+                try {
+                    $nuGetFeed = [NuGetFeed]::Create($feed.Url, $feed.Token, $feed.Patterns, $silent)
+                }
+                catch {
+                    Dump "Initiation of NuGetFeed failed. Error was $($_.Exception.Message)"
+                    continue

```

Comment:
Why `continue` here?

Reply:
Idea was that if you have 3 trusted feeds and one feed doesn't work - it continues to search for the package in other feeds instead of failing hard.
But, I actually think it should fail instead - will do that.